### ISSUE TITLE
DeleteExpression and VoidExpression are not translated to ESTree UnaryExpression

### ISSUE THREAD

**Issue by soda0289** [eslint/typescript-eslint-parser#171]
**What version of TypeScript are you using?**
2.2.1

**What version of `typescript-eslint-parser` are you using?**
2.0

**What code were you trying to parse?**

```ts
const a = {foo: 'bar'};
void 4;
delete a.foo;
```

**What did you expect to happen?**
Should generate ESTree node UnaryExpression with correct operator.

**What happened?**
Incorrect node is generated

Will post example later today. Should be simple fix.

**Comment by soda0289** [comment]
This causes the no-unused-expressions rule to throw false positives on delete expressions.

**Comment by JamesHenry** [comment]
Ah! I found this months ago and totally forgot about it. Thanks @soda0289! No example necessary

**Comment by soda0289** [comment]
Ill work on this today. Will submit PR soon
